Library/Homebrew/cmd/outdated.rb

```diff
@@ -16,6 +17,13 @@ def outdated_brews(formulae)
     formulae.map do |f|
       all_versions = []
       older_or_same_tap_versions = []
+
+      if f.oldname && !f.rack.exist?
+        if Pathname.new("#{HOMEBREW_CELLAR}/#{f.oldname}").exist?
+          raise Migrator::MigrationNeededError.new(f)
+        end
+      end
```

Comment:
There are some edge cases here as there is a chance that `HOMEBREW_CELLAR/f.oldname` is the rack of another formula.